jwql/website/apps/jwql/data_containers.py

```diff
@@ -146,6 +166,53 @@ def get_filenames_by_instrument(instrument):
     return filepaths
 
 
+def get_filenames_by_proposal(proposal):
+    """Return a list of filenames that are available in the filesystem
+    for the given ``proposal``.
+
+    Parameters
+    ----------
+    proposal : str
+        The five-digit proposal number (e.g. ``88600``).
+
+    Returns
+    -------
+    filenames : list
+        A list of filenames associated with the given ``proposal``.
+    """
+
+    filenames = sorted(glob.glob(os.path.join(
+        FILESYSTEM_DIR, 'jw{}'.format(proposal), '*')))
+    filenames = [os.path.basename(filename) for filename in filenames]
+
+    return filenames
+
+
+def get_filenames_by_rootname(rootname):
+    """Return a list of filenames available in the filesystem that
+    are part of the given ``rootname``.
+
+    Parameters
+    ----------
+    rootname : str
+        The rootname of interest (e.g. ``jw86600008001_02101_00007_guider2``).
+
+    Returns
+    -------
+    filenames : list
+        A list of filenames associated with the given ``rootname``.
+    """
+
+    proposal = rootname.split('_')[0].split('jw')[-1][0:5]
```

Comment:
@laurenmarietta I believe `filename_parser` can already handle rootnames, since it does `filename = os.path.basename(filename)` and the `basename` of a `filename` just returns the `filename`:

```python
In [2]: os.path.basename('some_jwst_filename')
Out[2]: 'some_jwst_filename'
```

However, since this service can accept any shortened version of the rootname, `filename_parser` will crash if you were to just give it something like `jw86600008001_02101`.  So unless we make `filename_parser` more robust, I think we will have to explicitly scrape out he proposal number until then.